commit b40f2ec52bf5647c4a1e71d73cd5a37a650f2944
Author: Maor Gottlieb <maorg@mellanox.com>
Date:   Sun Jul 1 15:50:17 2018 +0300

    IB/mlx5: Fix GRE flow specification
    
    [ Upstream commit a93b632c4531ff50c43d658447a45cbc11f488fd ]
    
    Currently the driver sets the mask of the gre_protocol to 0xffff
    without consideration in the user request.
    
    Fix it by copy the mask from the verbs spec.
    
    Fixes: da2f22ae7707 ("IB/mlx5: Add support for GRE flow specification")
    Signed-off-by: Maor Gottlieb <maorg@mellanox.com>
    Reviewed-by: Ariel Levkovich <lariel@mellanox.com>
    Signed-off-by: Leon Romanovsky <leonro@mellanox.com>
    Signed-off-by: Jason Gunthorpe <jgg@mellanox.com>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/infiniband/hw/mlx5/main.c b/drivers/infiniband/hw/mlx5/main.c
index cbeae4509359..85677afa6f77 100644
--- a/drivers/infiniband/hw/mlx5/main.c
+++ b/drivers/infiniband/hw/mlx5/main.c
@@ -2699,7 +2699,7 @@ static int parse_flow_attr(struct mlx5_core_dev *mdev, u32 *match_c,
 			 IPPROTO_GRE);
 
 		MLX5_SET(fte_match_set_misc, misc_params_c, gre_protocol,
-			 0xffff);
+			 ntohs(ib_spec->gre.mask.protocol));
 		MLX5_SET(fte_match_set_misc, misc_params_v, gre_protocol,
 			 ntohs(ib_spec->gre.val.protocol));
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1127_b40f2ec52bf5647c4a1e71d73cd5a37a650f2944.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1127_b40f2ec52bf5647c4a1e71d73cd5a37a650f2944.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
